test/controllers/tasks_controller_test.rb

```diff
@@ -83,42 +79,193 @@
       must_respond_with :redirect
       must_redirect_to task_path(new_task.id)
     end
+
+    it "will render :edit if attempts to enter an empty task are made" do
+      # Arrange
+      empty_task = {task: {name:nil, description:nil}}
```

Comment:
well-written tests! I think they read really well because your Arrange step always makes it clear how it's "invalid" if it is